app/models/katello/content_view.rb

```diff
@@ -579,14 +579,16 @@ def check_docker_repository_names!(environments)
 
     def check_composite_action_allowed!(env)
       if composite? && Setting['restrict_composite_view']
-        # verify that the composite's component view versions exist in the target environment.
+        env_ids = env.try(:pluck, 'id') || []
```

Comment:
why the `try(:pluck, 'id')`? instead of `env.pluck(:id)`?

Isn't it a safe assumption that the KTEnvironment AR relation has an id?

Reply:
So I kind of had to remind myself why I did this. :smile: ..env here can be a KTEnvironment AR relation or an array of envs. ```pluck``` will not work for a single KTEnvironment AR relation. Hence the try. And in case, it's a single AR relation, the next line should handle it.

Reply:
gotcha, its strange that this method is meant to handle varied input like that, but I guess its for a reason :man_shrugging: